### ISSUE TITLE
Confusing compiler error when you write `auto mut a`

### ISSUE THREAD

**Issue by rsaarelm** [Neat-Lang/neat#69]
```
    auto mut a = 0;
         ^ semicolon expected
```

The compiler should tell you you need to have `mut auto` instead.

**Comment by FeepingCreature** [comment]
Yep, the parser just steps through the attributes in a fixed order right now. I think it might actually view it as a function call as it's failing to parse it as a variable declaration. Needs fixing ofc.
